lib/helper/uploadAgentHelper.py

```diff
@@ -63,16 +64,20 @@ def upload_register_data(self, input_suite_fp, test_type):
                     tmp_list = case_full_path.split(os.sep)
                 else:
                     tmp_list = case_full_path.split(".")
-                if tmp_list[2].lower() == "t15":
-                    suite_name = "t15_summary_" + tmp_list[3].split("_")[1]
+                browser_type = tmp_list[3].split("_")[1].lower()
+                app_name = tmp_list[2].lower()
+                t_type = tmp_list[1].lower()
+                c_name = tmp_list[3].lower()
+                if browser_type not in upload_data:
+                    upload_data[browser_type] = {}
+                if app_name == "t15":
+                    suite_name = "t15_summary"
                 else:
-                    suite_name = tmp_list[1] + "_" + tmp_list[2] + "_" + tmp_list[3].split("_")[1]
-                if suite_name not in upload_data:
-                    upload_data[suite_name] = []
-                if tmp_list[3].split("_")[1] == "firefox":
-                    upload_data[suite_name].append(tmp_list[3])
+                    suite_name = t_type + "_" + app_name
+                if suite_name in upload_data[browser_type]:
+                    upload_data[browser_type][suite_name].append(c_name)
                 else:
-                    upload_data[suite_name].append(tmp_list[3])
+                    upload_data[browser_type] = {suite_name: [c_name]}
```

Comment:
If there are more than one suite under the same browser_type, it will be replaced.
`upload_data[browser_type] = {suite_name: [c_name]}` 
=> `upload_data[browser_type][suite_name] = [c_name]`